### PR TITLE
chore: use osm.tracingAddress in preset-mesh-config

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

Removed `tracing.address` field from values.yaml. Tracing address should be generated by `osm.tracingAddress` helper function.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [X] |

**Testing done**:

CI test

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

### CODE REVIEW

**Review by michelleN — COMMENTED** [review]
It looks to me like we should keep the helper function and use it in the templates instead of getting rid of it. Currently, the default tracing address would be incorrect if the user installed the osm control plane in any namespace other than osm-system without manually changing the tracing.address value in the values.yaml file. The template helper function allows us a better more automated configuration experience than modifying by the address in the yaml file by hand or passing it in through a flag.

**Review by snehachhabria — CHANGES_REQUESTED** [review]
+1 on @michelleN comment. I do not think we should be removing the helper function

**Comment by shashankram** [comment]
> +1 on @michelleN comment. I do not think we should be removing the helper function

My bad, I created the issue directly from a Github comment and it didn't convey the correct fix. I have updated the bug title and description: https://github.com/openservicemesh/osm/issues/3953

File: charts/osm/values.yaml [hunk 1]
```diff
@@ -187,10 +187,10 @@ OpenServiceMesh:
   tracing:
     # -- Toggles Envoy's tracing functionality on/off for all sidecar proxies in the mesh
     enable: false
-    # -- Address of the tracing collector service (must contain the namespace). When left empty, this is computed in helper template to "jaeger.<osm-namespace>.svc.cluster.local". Please override for BYO-tracing as documented in tracing.md
-    address: "jaeger.osm-system.svc.cluster.local"
     # -- Port of the tracing collector service
     port: 9411
+    # -- Address of the tracing collector service (must contain the namespace). When left empty, this is computed in helper template to "jaeger.<osm-namespace>.svc.cluster.local". Please override for BYO-tracing as documented in tracing.md
```
Comment (snehachhabria) [inline charts/osm/values.yaml:192]: nit: you can revert the change in this file

Comment (allenlsy) [reply charts/osm/values.yaml:192]: I found I cannot revert it. `address` somehow is a required field. Removing it will fail Chart checks

Comment (snehachhabria) [reply charts/osm/values.yaml]: looks like you were able to revert. I was not asking to remove the address field. The change you had just moved the lines

**Review by snehachhabria — APPROVED** [review]
Lets please link the issue this PR is resolving before merging

**Review by michelleN — DISMISSED** [review]
The default value for OpenServiceMesh.tracing.address should be left empty: `""` so that the namespace doesn't get hardcoded in the string and the helper can inject the namespace. https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml#L191

The documentation for the chart (charts/osm/README.md) should also be updated: https://github.com/openservicemesh/osm/blob/main/charts/osm/README.md

**Review by shashankram — APPROVED** [review]
